File: task1/Go/task1.go
package main

import (
	"fmt"
)

func main() {
	var N int
	fmt.Print("Enter N: ")
	fmt.Scan(&N)
	fmt.Print(closestPrimeNumber(N))
}

func isPrime(num int) bool {
	if num <= 1 {
		return false
	} // Числа меньше или равные 1 не являются простыми.
	for i := 2; i <= num/2; i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

func closestPrimeNumber(num int) int {
	if isPrime(num) {
		return 0
	}
	for i := 1; ; i++ {
		if isPrime(num-i) || isPrime(num+i) {
			return i
		}
	}
}

File: task1/Kotlin/task1.kt
//kotlin task1.kt -include-runtime -d task1.jar
//java -jar task1.jar
import kotlin.math.sqrt

fun isPrime(num: Int): Boolean {
    if (num <= 1) return false
    for (i in 2..sqrt(num.toDouble()).toInt()) {
        if (num % i == 0) return false
    }
    return true
}

fun closestPrimeNumber(num: Int): Int {
    if (isPrime(num)) return 0
    var i = 1
    while (true) {
        if (isPrime(num - i) || isPrime(num + i)) return i
        i++
    }
}

fun main() {
    print("Enter an integer: ")
    val N = readLine()!!.toInt()
    println(closestPrimeNumber(N))
}
